### ISSUE TITLE
10$: Suggest computed property on template literal as object key

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1082]
The following is invalid:
```javascript
let bar = 'bar';
let baz = 'baz';
let o = { 
  `foo`: true,
  `x${bar}`: true,
  `${baz}`: true,
};
```
We should diagnose the template keys and suggest using a computed property instead:
```javascript
let bar = 'bar';
let o = { 
  [`foo`]: true,
  [`x${bar}`]: true,
  [baz]: true,
};
```

**Comment by LeeWannacott** [comment]
Extra credit:
![image](https://github.com/quick-lint/quick-lint-js/assets/49783296/62ae5100-c6d2-4a79-9a8b-17a81bf8c456)
![image](https://github.com/quick-lint/quick-lint-js/assets/49783296/5318cace-ebb5-4265-9ba2-6ad7e8c1dc5e)
![image](https://github.com/quick-lint/quick-lint-js/assets/49783296/e8088a2f-1bad-4286-bbdd-a37fc91062a8)

**Comment by qayyum22** [comment]
i would like to work on it , @strager can you please assign it to me ?

**Comment by strager** [comment]
> can you please assign it to me ?

Sure!

If you want payment, make sure you follow the instructions on https://quick-lint-js.com/hiring/.
